File: solutions/ruby/bob/1/bob.rb
module Bob
  def self.hey(str0)
    str = str0.strip
    return "Fine. Be that way!" if str =~ /\A\s*\z/
    all_caps = (str =~ /\p{L}/) && (str == str.upcase)
    question = str.chars.last=="?"
    if all_caps and question
      "Calm down, I know what I'm doing!"
    elsif all_caps
      "Whoa, chill out!"
    elsif question
      "Sure."
    else
      "Whatever."
    end
  end
end

File: solutions/python/bob/1/bob.py
def response(hey_bob0):
    if hey_bob0 == '' or hey_bob0.isspace():
        return 'Fine. Be that way!'
    hey_bob = hey_bob0.strip()
    if hey_bob.upper() == hey_bob and hey_bob.lower() != hey_bob.upper():
        if hey_bob.endswith('?'):
            return 'Calm down, I know what I\'m doing!'
        return 'Whoa, chill out!'
    if hey_bob.endswith('?'):
        return 'Sure.'
    return 'Whatever.'
